oauthlocallib.php

```diff
@@ -257,7 +257,8 @@ public function build_signature(&$request, $consumer, $token) {
         $ok = openssl_sign($base_string, $signature, $privatekeyid);
 
         // Release the key resource
-        openssl_free_key($privatekeyid);
+        if(PHP_VERSION_ID < 80000)
+            openssl_free_key($privatekeyid); // this function is deprecated from PHP v8.0 onwards, no manual cleanup required - handles by PHP garbage collector
```

Comment:
Accoding to the PHP doco, the use of `openssl_free_key` is highly discourged. So I think we should mention this in the ticket and discuss the supported environments (e.g. PHP version and Moodle version) .